targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K24F/TARGET_MCU_K24F1M/device/system_MK24F12.h

```diff
@@ -112,8 +112,11 @@ extern "C" {
 #endif
 
 /* Define clock source values */
-
+#ifdef TARGET_WUNDERBAR
```

Comment:
Can this be achieved somehow else? any update to this header file will results in runtime errors. Can we keep sdk/cmsis files as they are and provide different solution to this?

@mmahadevan108

Reply:
I am all for setting this internally for Wunderbar target, but system_MK24F12.c/h is common for all MK24F boards and does not include anything board specific. This value has to be adjusted per board for correct SystemCoreClock setting.

Reply:
CPU_XTAL_CLK_HZ  macro is not used anymore in the SDK, everything. This is replaced by BOARD_XTAL0_CLK_HZ available in fsl_clock_config.h

Reply:
That's true - BOARD_XTAL0_CLK_HZ is used for clock settings, but SystemCoreClockUpdate uses CPU_XTAL_CLK_HZ and without #ifdef TARGET_WUNDERBAR SystemCoreClock value is completely wrong.

Reply:
I don't see this function invoked by the SDK. The SystemCoreClock is setup inside fsl_clock_config.c

Reply:
I'm sorry - you are correct. Problem with wrong SystemCoreClock was in my outdated app I was testing with. Changes removed.